drop obsolete recommendation of 1024-bit DHE as secure

davegarrett opened the pull request:
A newer recommendation could be put in as a replacement, however I don't think the sentence adds anything to the explanation, nor do I expect whatever could be recommended to hold up forever.

Also fix some punctuation in the same paragraph.

ekr commented on draft-ietf-tls-tls13.md:
```diff
@@ -1437,15 +1437,12 @@ Note that higher layers should not be overly reliant on whether TLS always
-a 1024-bit DHE key exchange with a host whose certificate you have
-verified, you can expect to be that secure.
+what they require.
 
```
I think it's good to keep an example here. Suggest you supply one with a different key size.

davegarrett commented:
Ok, I've added a new security recommendation. I'm listing a TODO with it, though, as Curve25519 is probably what we're going with as MTI for a reasonable expectation of security and there's nothing to cite yet. I don't think we'd want P256 cited in there for the baseline recommendation. This is fine until the MTIs are updated, I think. People would probably say 2048 DHE is fine, which for now it is, but I went with 3072 to match the same security level for the curve. Might need to discuss on list.

ekr commented:
I actually expect P256 to be the MTI based on the discussion in person and on-list.

Given that RSA 2048 seems to be what browsers are pushing, I suspect that that's
what's needed here.

davegarrett commented:
I essentially have no problem with that currently, but this spec will be around for quite a while. I'd rather have the new general purpose expectations in the new spec so that this will age better. (you could argue I'm overthinking this, though) I could also just drop the DHE and list only ECDHE.

ekr commented:
I'm fine with changing this to 256-bit EC keys, but note that you'll want to refer to that for signatures as well.

davegarrett commented:
Ok. I've switched it to just 255 bit ECDHE and added 224 bit ECDSA as the minimum signature. (arguably the latter could be higher, but we went with 112 bit symmetric in the backwards compatibility restrictions) I'm specifically going with 255 instead of 256 so Curve25519 isn't ruled out by a bit.
